### PR TITLE
Documentation

### PR DESCRIPTION
**Pull request by sidraval**
I made a mess out of this! Here's the original https://github.com/thoughtbot/Swish/pull/75

### CODE REVIEW

File: Documentation/Advanced.md [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+- Define a custom `ResponseParser` to provide to our `Request` object, because
+the default one parses `JSON`.
+- Define a custom `Deserializer` to provide to our `APIClient`, because the 
+default one uses `NSJSONSerialization`.
+
+We'll start with the custom `Deserializer`. It will simply pass along the 
+`NSData` if it exists:
+
+```swift
+struct StringDeserializer: Deserializer {
```
Comment (gfontenot) [inline Documentation/Advanced.md]: how about calling this `DataDeserializer`?

File: Documentation/Advanced.md [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+let client = APIClient(
+  requestPerformer: NetworkRequestPerformer(),
+  deserializer: StringDeserializer()
+)
+let request = AuthorizedRequest()
+
+let dataTask = client.performRequest(request) { (result: Result<Bool, SwishError>) in
+  // whatever you need to do
+}
+```
```
Comment (gfontenot) [inline Documentation/Advanced.md:88]: What do you think about ending with a note that you could also simplify this by creating a boolean parser using the same `DataDeserializer` but leave it up as an exercise for the reader?

**Comment by gfontenot** [comment]
Couple of small notes, but this all reads nicely to me. Maybe cc @thoughtbot/copy-editing

File: Documentation/Advanced.md (hunk 1 above)
Comment (sidraval) [inline Documentation/Advanced.md]: 👍

File: Documentation/Advanced.md (hunk 2 above)
Comment (sidraval) [inline Documentation/Advanced.md:88]: 👍

**Pull request merged**